### ISSUE TITLE
Font codes stripped in dist/ css file.

### ISSUE THREAD

**Issue by nerdybeast** [martndemus/ember-font-awesome#107]
All of the fonts in my app are being rendered as little boxes.  I thought at first perhaps the font-awesome css wasn't making into the final app css file in the `dist/` folder but to my surprise, all of the css is there but the font codes have been stripped away.  Here is my setup:

ember: 2.6.0
ember-cli: 2.4.0
font-awesome: 4.5.0
ember-font-awesome: 2.2.0

I completed these steps to install:

ran `ember install ember-font-awesome`

updated `ember-cli-build.js` to include:
```
'ember-font-awesome': {
    useScss: true
}
```
added `@import "font-awesome";` in `app.scss`

stopped & restarted the server

but here's where it gets weird (I think)... This is an ACTUAL snippet from my app's css file in the `dist/` directory (in my case it's `dist/assets/sith.css`):
```
.fa-plus-circle:before {
  content: ""; }

.fa-minus-circle:before {
  content: ""; }

.fa-times-circle:before {
  content: ""; }

.fa-check-circle:before {
  content: ""; }
```

Not sure how the little box encoding ended up there instead of the font codes that come from the original font-awesome source file.

Thoughts?

(Also I have tried removing the `ember-cli-build.js` change, removing the `@import "font-awesome"`, and going straight with an `app.import('bower_components/font-awesome/css/font-awesome.min.css')` statement and I get the same result.)

**Comment by nesevis** [comment]
Did you ever resolve this?

**Comment by nerdybeast** [comment]
No, but I didn't look into it any further after creating this ticket.

**Comment by OakBehringer** [comment]
Same problem here. Works locally when serving via ember-cli, but after deploying I get little boxes. BUT - I only get little boxes for FA 4.7 icons, to which I just upgraded. All the icons from the previous version that I had installed still work. Help.

**Comment by buschtoens** [comment]
Could everyone that is experiencing this issue save these files and re-upload them? When re-uploading, please don't copy and paste, but really *download* the file. Ideally just post the links to the deployed files as they are.

If these files contain confidential info feel free to mail me instead.

**Comment by OakBehringer** [comment]
The problem for me was either a cached font file in cloudfront, that wasn't invalidated via ember-cli-deploy, or the new font file wasn't uploaded to s3 (via ember-cli-deploy). I wiped my s3 and invalidated every file on CloudFront, and everything started to work. 

My guess is that the old font file was cached and being served, which didn't have the new FA icons. 

All fixed for me.

**Comment by buschtoens** [comment]
Thanks for the follow-up, @OakBehringer!

FWIW, `broccoli-asset-rev` [doesn't fingerprint font files](https://github.com/rickharrison/broccoli-asset-rev/blob/e5ee2bc70876015ff0c653409190b74831792d6f/lib/default-options.js#L5) by default. However, you can easily add the necessary extensions as described in the [ember-cli-deploy](http://ember-cli-deploy.com/docs/v0.6.x/fingerprinting/#usage) and [ember-cli](https://ember-cli.com/user-guide/#fingerprinting-and-cdn-urls) docs:

```js
// ember-cli-build.js

// ...
const app = new EmberApp({
  fingerprint: {
    extensions: ['js', 'css', 'png', 'jpg', 'gif', 'ico', 'map', 'svg', 'woff', 'woff2', 'otf', 'eot', 'ttf']
  }
});
// ...
```

**Comment by buschtoens** [comment]
@nerdybeast I just re-read your issue. I didn't notice your `little box encoding` remark the first time. My browser / OS combination (Chrome / Arch Linux) does not show a [replacement character](http://www.fileformat.info/info/unicode/char/fffd/index.htm) in your code snippet.

For me it looks like this:

![image](https://user-images.githubusercontent.com/834636/29701950-10191ae4-896f-11e7-84ce-0ee3b35b87f2.png)

While it probably should look something like this;

```css
.fa-plus-circle:before {
  content: "�"; } /* .charCodeAt(0).toString(16) => "f055" */

.fa-minus-circle:before {
  content: "�"; } /* .charCodeAt(0).toString(16) => "f056" */

.fa-times-circle:before {
  content: "�"; } /* .charCodeAt(0).toString(16) => "f057" */

.fa-check-circle:before {
  content: "�"; } /* .charCodeAt(0).toString(16) => "f058" */
```

These characters exactly match the codes in the original [`font-awesome.css`](https://github.com/FortAwesome/Font-Awesome/blob/cef4f1ef8c0ee2b551ce84c975739d55dc80f10f/css/font-awesome.css#L431-L442).

So as it turns out, the font codes are not *stripped*, but rather the escaped UTF-8 chars (`\f055`) are transformed into "real" UTF-8 characters. Theoretically this should not brick anything.

So there are two questions:

1. What transforms the escape sequences (and why)?
2. Why does it brick in your browser?

/cc @nesevis do you concur?

**Comment by buschtoens** [comment]
For 1. I would assume that either Sass, some other transpilation tool in your pipeline or some minification step thinks that it's a good idea to do that, probably to save space.

**Comment by OakBehringer** [comment]
Thanks buschtoens - that was a big help
